internal/controller/postgrescluster/controller.go

```diff
@@ -136,6 +136,15 @@ func (r *Reconciler) Reconcile(
 			return runtime.ErrorWithBackoff(err)
 		}
 	}
+	// Issue Warning Event if postgres version is EOL according to PostgreSQL:
+	// https://www.postgresql.org/support/versioning/
+	currentTime := time.Now()
+	if postgres.ReleaseIsFinal(cluster.Spec.PostgresVersion, currentTime) {
+		r.Recorder.Eventf(cluster, corev1.EventTypeWarning, "EndOfLifePostgresVersion",
+			"The last minor version of Postgres %v has been released. PG %v will no longer receive "+
+				"updates. We recommend upgrading. See https://www.postgresql.org/support/versioning/.",
+			cluster.Spec.PostgresVersion, cluster.Spec.PostgresVersion)
```

Comment:
⛏️ I recommend passing the version only once to `fmt`.
⛏️ I find these messages easier to read in the code when broken up between sentences.

⛏️ I worry that a program that turns URLs into links will slurp up the trailing dot. For some reason, I find trailing slashes in URLs unappealing, too.

```suggestion
			"The last minor version of Postgres %[1]v has been released." +
				" PG %[1]v will no longer receive updates. We recommend upgrading."+
				" See https://www.postgresql.org/support/versioning",
			cluster.Spec.PostgresVersion)
```